File: ide-plugin/src/org/jetbrains/plugins/ruby/ruby/codeInsight/SignatureService.java
package org.jetbrains.plugins.ruby.ruby.codeInsight;

import org.jetbrains.ruby.runtime.signature.server.SignatureServer;

import java.util.logging.Logger;


public class SignatureService {
    private static final Logger LOGGER = Logger.getLogger("SignatureService");
    public SignatureService() {
        new Thread() {
            @Override
            public void run() {
                SignatureServer server = SignatureServer.getInstance();
                try {
                    server.runServer();
                } catch (Exception e) {
                    LOGGER.severe(e.getMessage());
                    e.printStackTrace();
                }

            }
        }.start();
    }

}

File: ide-plugin/src/org/jetbrains/plugins/ruby/ruby/codeInsight/SignatureService.kt
package org.jetbrains.plugins.ruby.ruby.codeInsight

import com.intellij.openapi.diagnostic.Logger
import org.jetbrains.plugins.ruby.ruby.persistent.TypeInferenceDirectory
import org.jetbrains.ruby.codeInsight.types.storage.server.DatabaseProvider
import org.jetbrains.ruby.runtime.signature.server.SignatureServer
import java.io.File


class SignatureService {
    init {
        DatabaseProvider.connect(filePath = File(TypeInferenceDirectory.RUBY_TYPE_INFERENCE_DIRECTORY.toFile(), "ruby-type-inference").absolutePath)
        DatabaseProvider.createAllDatabases()

        Thread {
            while (true) {
                try {
                    SignatureServer.runServer()
                } catch (e: Exception) {
                    LOGGER.error(e)
                }

            }
        }.start()
    }

    companion object {
        private val LOGGER = Logger.getInstance("SignatureService")
    }

}